lib/apr/notifications.ex

```diff
@@ -3,69 +3,70 @@ defmodule Apr.Notifications do
   require Logger
 
   def receive_event(event, topic, routing_key) do
-    event
-    |> slack_message(topic, routing_key)
-    |> post_message(topic, routing_key)
+    get_subscriptions(topic, routing_key)
+    |> Enum.map(&{&1, slack_message(&1, event, topic, routing_key)})
+    |> Enum.map(&post_message(&1))
   end
 
-  def slack_message(event, topic_name, routing_key) do
+  defp get_subscriptions(topic, routing_key) do
+    Subscriptions.get_topic_subscribers(topic, routing_key)
+  end
+
+  def slack_message(subscription, event, topic_name, routing_key) do
     with topic when not is_nil(topic) <- Subscriptions.get_topic_by_name(topic_name) do
       case topic.name do
         "subscriptions" ->
-          Apr.Views.SubscriptionSlackView.render(event)
+          Apr.Views.SubscriptionSlackView.render(subscription, event)
 
         "inquiries" ->
-          Apr.Views.InquirySlackView.render(event)
+          Apr.Views.InquirySlackView.render(subscription, event)
 
         "purchases" ->
-          Apr.Views.PurchaseSlackView.render(event)
+          Apr.Views.PurchaseSlackView.render(subscription, event)
 
         "auctions" ->
-          Apr.Views.BiddingSlackView.render(event)
+          Apr.Views.BiddingSlackView.render(subscription, event)
 
         "radiation.messages" ->
-          Apr.Views.RadiationMessageSlackView.render(event)
+          Apr.Views.RadiationMessageSlackView.render(subscription, event)
 
         "conversations" ->
-          Apr.Views.ConversationSlackView.render(event)
+          Apr.Views.ConversationSlackView.render(subscription, event)
 
         "invoices" ->
-          Apr.Views.InvoiceSlackView.render(event, routing_key)
+          Apr.Views.InvoiceSlackView.render(subscription, event, routing_key)
 
         "consignments" ->
-          Apr.Views.ConsignmentsSlackView.render(event)
+          Apr.Views.ConsignmentsSlackView.render(subscription, event)
 
         "feedbacks" ->
-          Apr.Views.FeedbacksSlackView.render(event)
+          Apr.Views.FeedbacksSlackView.render(subscription, event)
 
         "sales" ->
-          Apr.Views.SalesSlackView.render(event, routing_key)
+          Apr.Views.SalesSlackView.render(subscription, event, routing_key)
 
         "commerce" ->
-          Apr.Views.CommerceSlackView.render(event, routing_key)
+          Apr.Views.CommerceSlackView.render(subscription, event, routing_key)
 
         "partners" ->
-          Apr.Views.PartnersSlackView.render(event, routing_key)
+          Apr.Views.PartnersSlackView.render(subscription, event, routing_key)
       end
     else
       _ -> Logger.warn("Unknown Topic #{topic_name}")
     end
   end
 
-  defp post_message(slack_message, topic, routing_key) do
+  defp post_message({subscriber, slack_message}) do
     if slack_message != nil do
```

Comment:
i know this is not part of your PR, but maybe we should make this a guard statement on the function itself and have a separate catch all method that does nothing :)
```elixir
defp post_message({subscriber, slack_message}) when not is_nil(slack_message) do
...
end
defp post_message(_), do: nil
```